modules/api-gateway/main.tf

```diff
@@ -0,0 +1,227 @@
+# API Gateway- API for ORCA cumulus reconciliation
+resource "aws_api_gateway_rest_api" "orca_cumulus_reconciliation_api" {
+  name        = "${var.prefix}_orca_cumulus_reconciliation_api"
+  description = "API for catalog reporting, request_status_for_job and request_status_for_file lambda functions"
+  endpoint_configuration {
+    types = ["PRIVATE"]
+    # Cumulus might need to create vpc_endpoint_ids which should be added here in the future in order to access this API
+  }
+}
+
+resource "aws_api_gateway_rest_api_policy" "orca_cumulus_reconciliation_api_resource_policy" {
+  rest_api_id = aws_api_gateway_rest_api.orca_cumulus_reconciliation_api.id
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "*"
+      },
+      "Action": "execute-api:Invoke",
+      "Resource": "${aws_api_gateway_rest_api.orca_cumulus_reconciliation_api.execution_arn}"
+    },
+    {
+      "Effect": "Deny",
+      "Principal": {
+        "AWS": "*"
+      },
+      "Action": "execute-api:Invoke",
+      "Resource": "${aws_api_gateway_rest_api.orca_cumulus_reconciliation_api.execution_arn}",
+      "Condition": {
+        "StringNotEquals": {
+          "aws:SourceVpc": "${var.api_gateway_policy_vpc_id}"
+        }
+      }
+    }
+  ]
+}
+EOF
+}
+
+#API details for orca_catalog_reporting lambda
+resource "aws_api_gateway_resource" "orca_catalog_reporting_api_resource" {
+  path_part   = "orca_catalog_reporting"
```

Comment:
Should probably discuss this further. I think we would want a `path` instead of `path_part` and we would want to follow RESTFUL API standards.